source/Minecraft.cpp

```diff
@@ -452,7 +452,10 @@ void Minecraft::tickInput()
 			}
 			else if (keyCode == AKEYCODE_SEARCH)
 			{
-				m_options.field_23D ^= 1;
+				m_options.m_iPerspective++;
+				// If the perspective goes over the max loop back
+				if (m_options.m_iPerspective > 1) 
+					m_options.m_iPerspective = 0;
```

Comment:
Please revert this back to the binary xor, or use `x = !x;`. You can't actually `++` a boolean. Also it's probably `m_bThirdPerson`.

While you're at it, change `field_23D`/`m_iPerspective` to a bool. It's actually a bool, not a uint8_t, but I didnt bother changing it back.

Reply:
I was thinking that it would be more ideal to have it as an int and have it as perspective in the inevitable case anyone wants to implement the front-facing thirdperson perspective. Do you still want me to have it be a boolean or is that a reasonable assumption and change?

Reply:
fixed in https://github.com/ReMinecraftPE/mcpe/pull/41/commits/fa385f886455a6f0023ca7abb39467506d20214f